components/global/global_search_nav/global_search_nav.tsx

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+import Flex from '@mattermost/compass-components/utilities/layout/Flex';
+
+import Search from 'components/search';
+import QuickSwitchModal from 'components/quick_switch_modal';
+import {
+    Constants,
+    ModalIdentifiers,
+    RHSStates,
+} from 'utils/constants';
+
+import * as Utils from 'utils/utils';
+
+type Props = {
+    rhsState: typeof RHSStates[keyof typeof RHSStates] | null;
+    rhsOpen: boolean;
+    isQuickSwitcherOpen?: boolean;
+    actions: {
+        showMentions: () => void;
+        openRHSSearch: () => void;
+        closeRightHandSide: () => void;
+        openModal: (modalData: {modalId: string; dialogType: any; dialogProps?: any}) => void;
+        closeModal: (modalId: string) => void;
+    };
+};
+
+const GlobalSearchNav = (props: Props): JSX.Element => {
+    const {rhsState, rhsOpen, actions: {closeRightHandSide, showMentions}} = props;
+
+    useEffect(() => {
+        document.addEventListener('keydown', handleShortcut);
+        document.addEventListener('keydown', handleQuickSwitchKeyPress);
+    }, []);
+
+    const searchMentions = () => {
+        if (rhsState === RHSStates.MENTION) {
+            closeRightHandSide();
+        } else {
+            showMentions();
+        }
+    };
+
+    const handleShortcut = (e: KeyboardEvent) => {
+        const {actions: {closeModal}} = props;
+
+        if (Utils.cmdOrCtrlPressed(e) && e.shiftKey) {
+            if (Utils.isKeyPressed(e, Constants.KeyCodes.M)) {
+                e.preventDefault();
+                closeModal(ModalIdentifiers.QUICK_SWITCH);
+                searchMentions();
+            }
+            if (Utils.isKeyPressed(e, Constants.KeyCodes.L)) {
+                // just close the modal if it's open, but let someone else handle the shortcut
+                closeModal(ModalIdentifiers.QUICK_SWITCH);
+            }
+        }
+    };
+
+    const toggleQuickSwitchModal = () => {
+        const {isQuickSwitcherOpen, actions: {openModal, closeModal}} = props;
+
+        if (isQuickSwitcherOpen) {
+            closeModal(ModalIdentifiers.QUICK_SWITCH);
+        } else {
+            openModal({
+                modalId: ModalIdentifiers.QUICK_SWITCH,
+                dialogType: QuickSwitchModal,
+            });
+        }
+    };
+
+    const handleQuickSwitchKeyPress = (e: KeyboardEvent) => {
+        if (Utils.cmdOrCtrlPressed(e) && !e.shiftKey && Utils.isKeyPressed(e, Constants.KeyCodes.K)) {
+            if (!e.altKey) {
+                e.preventDefault();
+                toggleQuickSwitchModal();
+            }
+        }
+    };
+
+    return (
+        <Flex
```

Comment:
Seems strange to have a component for a flex container. Why not just a styled component?

Reply:
The idea for the flex utility is to avoid the need to create the styled components separately and nest divs, this container already gives us a flex parent.

Reply:
This particular component is an early/foundational implementation of a planned layout component that is intended to:

* make it easer for a dev to layout components without having to know/formulate CSS
* that inherently adheres to the Compass Design layout guidelines for consistency
* and removes the need to constantly (re)write the same spacing/padding CSS in styled components every time there is a need to space components in a consistent manner.

For example, if several icon buttons need to be laid out horizontally with pre-determined 8px spacing based on the Design System (for instance), a dev would be able to do something like this:

```javascript
<Row spacing={8}>
    <IconButton ... />
    <IconButton ... />
    <IconButton ... />
    <IconButton ... />
</Row>
```

Typescript would then suggest/enforce the pre-determined/approved spacing from the design system.

Flex may be the foundation element under the hood that makes something like this possible, but we're not quite there yet.

The one thing I like about styled-components is that you can have a component name that is more semantic to what its purpose is, which is something to consider as well. Though in that case, it would obviously be possible to do something this to help maintain consistency with the Design system and reduce repetition in code:

```javascript
const GlobalSearchNav = styled(Row)``;
```

(Not sure off hand if the back ticks are needed here)